### PR TITLE
[prometheus-adapter] create pod disruption budget

### PR DESCRIPTION
**Pull request by ahmednasir91**
#### What this PR does / why we need it:
Adds support for creating pod disruption budget when its enabled.

#### Checklist
- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by monotek — APPROVED** [review]

File: charts/prometheus-adapter/templates/custom-metrics-apiserver-pdb.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+spec:
+  {{- if .Values.podDisruptionBudget.minAvailable }}
+  minAvailable: {{ .Values.podDisruptionBudget.minAvailable }}
+  {{- end  }}
+  {{- if .Values.podDisruptionBudget.maxUnavailable }}
+  maxUnavailable: {{ .Values.podDisruptionBudget.maxUnavailable }}
+  {{- end  }}
+  selector:
+    matchLabels:
+      app: {{ template "k8s-prometheus-adapter.name" . }}
```
Comment (steven-sheehy) [inline charts/prometheus-adapter/templates/custom-metrics-apiserver-pdb.yaml:20]: This should include the release label like the selectors in the service and deployment.

File: charts/prometheus-adapter/values.yaml [hunk 2]
```diff
@@ -143,3 +143,10 @@ hostNetwork:
   # API server unable to communicate with metrics-server. As an example, this is required
   # if you use Weave network on EKS
   enabled: false
+
+podDisruptionBudget:
+  # Specifies if PodDisruptionBudget should be enabled
+  # When enabled, minAvailable or maxUnavailable should also be defined.
+  enabled: false
+  minAvailable:
```
Comment (steven-sheehy) [inline charts/prometheus-adapter/values.yaml:151]: Would be good to specify defaults for these both for recommendations and to avoid nils.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: charts/prometheus-adapter/templates/custom-metrics-apiserver-pdb.yaml (hunk 1 above)
Comment (ahmednasir91) [reply charts/prometheus-adapter/templates/custom-metrics-apiserver-pdb.yaml:20]: Added release label.

File: charts/prometheus-adapter/values.yaml (hunk 2 above)
Comment (ahmednasir91) [reply charts/prometheus-adapter/values.yaml:151]: Added default for `maxUnavailable` and have condition for `minAvailable` So there will be no nils.

```yaml
# Source: prometheus-adapter/templates/custom-metrics-apiserver-pdb.yaml
apiVersion: policy/v1beta1
kind: PodDisruptionBudget
metadata:
  name: RELEASE-NAME-prometheus-adapter
  labels:
    app: prometheus-adapter
    chart: prometheus-adapter-2.7.0
    release: RELEASE-NAME
    heritage: Helm
spec:
  maxUnavailable: 1
  selector:
    matchLabels:
      app: prometheus-adapter
      release: RELEASE-NAME
```

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**